### DISCUSSION TITLE
Let user update JSON file

### DISCUSSION THREAD

**Discussion by BLVANSONHVA** [cables-gl/cables_docs#769] [Answers & Questions]
<img width="782" alt="Screenshot 2024-05-22 at 12 25 07" src="https://github.com/cables-gl/cables_docs/assets/156310768/fae3dfaf-479f-4dbd-837d-20191ad0fdc7">
<img width="188" alt="Screenshot 2024-05-22 at 12 25 17" src="https://github.com/cables-gl/cables_docs/assets/156310768/ab628a7f-13e8-496a-92d0-667e5e08754d">
Dear community,

I am using Cables.gl to prototype a tool that is basically an interactive data visualisation. The user gets presented with a data visualisation based on a JSON file. I wish that the user can input something that then gets added to the JSON file, and therefore represented in the data visualisation. 

I have included two screenshots. The first is the data visualisation (using EchartsEvents) and the second is a div element where the user can input. I hope that this input can be injected back into the JSON somehow, and therefore reflected in the datavis.

**Comment by steam0r** [comment]
you can get the data from the elment and then use ObjectSet... to manipulate the json that is fed into echarts

in this example patch you'd put your object manipulation below the "ParseObject" on the top right:

https://cables.gl/edit/iY6lbI

this is list of ops to manipulate json-data in cables:

https://cables.gl/ops/Ops.Json

**Reply by BLVANSONHVA** [reply]
Thanks for your reply! 
I understand what you mean, but I do not know how to do it. Is there any simple example available ?

**Reply by BLVANSONHVA** [reply]
More specifically, how do you determine the place in the JSON file that you insert an object in? 
If we take the example file you sent, then let's say you wish to create another entity in the data on the x-axis. How could you approach that?

**Reply by steam0r** [reply]
this is more of an echarts question, cables cannot know how echarts expects it's data and i think it's different from chart to chart
